.github/workflows/ci-tests.yml

```diff
@@ -0,0 +1,138 @@
+########################
+## GitHub Actions CI  ##
+########################
+#
+# Our preferred CI usage is Circle CI
+# but this acts as alternative if not available
+#
+name: "Run CI Tests"
+on:
+  workflow_dispatch:
+  pull_request:
+    paths-ignore:
+      - '**.png'
+      - '**.md'
+      - '**.rst'
+jobs:
+  linting:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        jobs: [ 'linting', 'doclinting' ]
+    name: ${{ matrix.jobs }} tests
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.9'
+    - name: Install dependencies
+      run: |
+        pip install -Ur requirements.txt -Ur requirements_dev.txt
+        python setup.py develop
+        pip install tox
+    - name: Set Tox Params
+      run: |
+        echo "TOXENV=${{ matrix.jobs }}" >> $GITHUB_ENV
+        echo "UPLOAD_COVERAGE=0" >> $GITHUB_ENV
+    - name: Run the tests
+      run: |
+        tox -e ${{ env.TOXENV }}
+  python-version-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ '3.6', '3.7', '3.8', '3.9' ]
```

Comment:
Want to consider fewer versions here as you mentioned elsewhere?

Reply:
I think I'd like to keep it as is for now, in case there's any version specific things in there (either now or in future). I see 3.6 has different parallel processing for example.

So let's leave this as a bigger discussion. And maybe it's not an issue if we can get on CircleCI as an open source project.